ibm/data_source_ibm_pi_placement_group.go

```diff
@@ -0,0 +1,79 @@
+// Copyright IBM Corp. 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	st "github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/helpers"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceIBMPIPlacementGroup() *schema.Resource {
+
+	return &schema.Resource{
+		ReadContext: dataSourceIBMPIPlacementGroupRead,
+		Schema: map[string]*schema.Schema{
+			helpers.PIPlacementGroupID: {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			helpers.PIPlacementGroupName: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			helpers.PIPlacementGroupPolicy: {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			helpers.PICloudInstanceId: {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			PIPlacementGroupMembers: {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceIBMPIPlacementGroupRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	sess, err := meta.(ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	powerinstanceid := d.Get(helpers.PICloudInstanceId).(string)
+	placementGroupid := d.Get(helpers.PIPlacementGroupID).(string)
+	client := st.NewIBMPIPlacementGroupClient(ctx, sess, powerinstanceid)
+
+	response, err := client.Get(placementGroupid)
```

Comment:
You should be passing the name instead of the id.

Reply:
I made this suggested change.